app/(app)/room/[roomId].tsx

```diff
@@ -0,0 +1,80 @@
+import { Text, View, StyleSheet, FlatList, TextInput } from "react-native";
+import { useLocalSearchParams } from "expo-router";
+import React, { useEffect, useState } from "react";
+import {
+  FieldValue,
+  addDoc,
+  doc,
+  onSnapshot,
+  updateDoc,
+} from "firebase/firestore";
+import { firestoreDB } from "../../../firebaseConfig";
+import { useAuthStore } from "../../../stores/auth";
+
+type Message = {
+  username: string;
+  message: string;
+};
+
+export default function Room() {
+  const local = useLocalSearchParams();
+  const [messages, setMessages] = useState<Array<Message>>([]);
+  const auth = useAuthStore();
+  console.log(auth.user);
+
+  const sendMessage = (message: string) => {
+    updateDoc(doc(firestoreDB, `rooms/${local.roomId}/`), {
+      messages: [...messages, { username: auth.user?.email, message }],
+    });
+  };
+
+  useEffect(() => {
+    if (typeof local.roomId !== "string") return;
+    const unsub = onSnapshot(
+      doc(firestoreDB, `rooms/${local.roomId}`),
+      (doc) => {
+        setMessages(doc.data()?.messages);
+      }
+    );
+    return unsub;
+  }, []);
```

Comment:
Wygląda iż jak dwie osoby wyślą wiadomość jednocześnie to się zrobi race condition, ale działa